### ISSUE TITLE
Sign provenance files

### ISSUE THREAD

**Issue by marcofranssen** [philips-labs/slsa-provenance-action#92]
Signing of provenance files should allow for compatibility with:

- [ ] [in-toto](https://github.com/in-toto/attestation)
- [ ] [cosign](https://github.com/sigstore/cosign)
- [ ] [notation](https://github.com/notaryproject/notation)

Both Cosign and notation allow for attaching additional artefacts to the image. Eventually this signed provenance should also be attachable to the container image (#88). That would allow us to fetch the provenance for a given docker image and use the tooling (notation/cosign) to verify the signature.

**Comment by pieterlexis** [comment]
I'm using the [DSSE](https://github.com/secure-systems-lab/dsse/blob/master/protocol.md) signing implementation of [sigstore](https://github.com/sigstore/sigstore) in #91 which indeed generates the payload and signatures as defined in [in-toto](https://github.com/in-toto/attestation/blob/main/spec/README.md#envelope).

I think the integration of the generated, signed provenance is beyond the scope of what I intended to implement in #91.

**Comment by marcofranssen** [comment]
@pieterlexis yes integrating the attaching to docker images or auto uploading to Github releases can be done from separate PR.

**Comment by ChaosInTheCRD** [comment]
I would be interested in working on this PR ✋ 😄

**Comment by ChaosInTheCRD** [comment]
For Docker Images, I have been looking at the implementation within Tekton Chains... and I can see it could be possible to do something similar here too

https://github.com/tektoncd/chains/blob/a7e0557bf10abbe1d4f3f90e947857ba87890194/pkg/chains/storage/oci/oci.go#L100

I wonder if this should be integrated as part of the `sign-provenance` action that @pieterlexis has written. In an ideal world, you would be able to have all of this completed within a single command, like:
- Declare the repo, digest, tags etc.
- Generate Provenance
- Sign the Provenance
- Push the Provenance to the remote

That way, you can be sure that the attestation has not been modified / compromised before it makes it to the registry.

I wonder if it would be worthwhile for me to expand on https://github.com/philips-labs/slsa-provenance-action/pull/88 so someone can just call a flag that allows a user to achieve all of this for a docker image. As far as I can see with SLSA, this is a common flow for anyone wanting to generate Level 2 Provenance for an image.

**Comment by marcofranssen** [comment]
@ChaosInTheCRD what you have shared in your workflow tryout https://github.com/ChaosInTheCRD/mic-test/runs/4514052305?check_suite_focus=true is exactly what we would like to achieve once this PR and #88 are merged in a single line command. @pieterlexis could you also have a look at @ChaosInTheCRD his workflow we can achieve this with the combination of both PR's?
